Add debug/watch/context to list of valid menu extension points

adrianstephens opened the pull request:
<!-- Thank you for submitting a Pull Request. Please:
* Read our Pull Request guidelines:
  https://github.com/microsoft/vscode/wiki/How-to-Contribute#pull-requests
* Associate an issue with the Pull Request.
* Ensure that the code is up-to-date with the `main` branch.
* Include a description of the proposed changes and how to test them.
-->
This is a fix for #237750 - it just adds debug/watch/context to the list of valid menu extension points, which I presume was accidentally missed.

adrianstephens committed:
debug/variables/context

gjsjohnmurray commented:
Does the title of this PR need correcting?

sanqro reviewed (approved):
:+1:

adrianstephens committed:
Merge branch 'main' into enable-debug/watch/context-menu-contribution

connor4312 commented:
Yea this should be fine, we already have similar menu extension points for other places variables/expressions are shown

connor4312 committed:
Merge branch 'main' into enable-debug/watch/context-menu-contribution

roblourens reviewed (changes requested):
Did anybody try this?

<img width="488" alt="image" src="https://github.com/user-attachments/assets/821b2b6c-e0b5-48bb-a755-dd51a02eea94" />

The hard part about this is what the toolbar context should be. We need to define a new object which is safe to serialize and has a reasonable shape, and then that object is basically API

adrianstephens committed:
Merge branch 'microsoft:main' into enable-debug/watch/context-menu-contribution

adrianstephens committed:
watchExpressionsView context fixes

adrianstephens commented:
I had a stab at the watch view context.  In the interests of doing the least harm I added a toJSON to the Expression class which serializes it to the same thing the Variables view does - basically a session id and a DAP Variable. The Variables view seems to work around the same problem by using a global variable 'variableInternalContext'. I like my idea better!
While I was in there, I moved the data breakpoint commands to debugCommands.ts, and added them to the Watch view.

adrianstephens committed:
types in context; drag-n-drop watch

adrianstephens committed:
CONTEXT_VARIABLE_TYPE for Variable and Watch contexts

thegecko commented:
Does this duplicate https://github.com/microsoft/vscode/pull/212501

adrianstephens commented:
@thegecko - huh; it looks like it does - although I'm only trying to bring the Watch panel to parity with Variables. Sorry I didn't notice yours.
Do you want to review mine and/or add to it? Do you have any insight into why yours went no further, or how we can prevent that happening to this one?

adrianstephens commented:
@roblourens is there any realistic chance of you looking at this, or should I abandon it?

adrianstephens committed:
Merge branch 'main' into enable-debug/watch/context-menu-contribution

connor4312 commented:
thanks!
